radix-engine-interface/src/blueprints/component.rs

```diff
@@ -0,0 +1,157 @@
+use core::marker::PhantomData;
+
+use radix_engine_common::prelude::*;
+
+pub trait TypeInfoMarker {
+    const PACKAGE_ADDRESS: Option<PackageAddress>;
+    const BLUEPRINT_NAME: &'static str;
+    const OWNED_TYPE_NAME: &'static str;
+    const GLOBAL_TYPE_NAME: &'static str;
+}
+
+pub struct Global<T>(pub ComponentAddress, PhantomData<T>)
```

Comment:
I feel like these should maybe be defined in `radix-engine` or `native-sdk` to save them conflicting with the scrypto types?

Reply:
The thing is that we need to use `Global<T>` and `Owned<T>` in the `radix-engine-interface` crate where all of the invocation structs and types are defined and the `radix-engine-interface` does not depend on the `radix-engine` or the `native-sdk`, quite the opposite: they depend on it. So, defining this type there would result in a circular dependency.